src/Symfony/Component/DependencyInjection/Loader/FileLoader.php

```diff
@@ -101,21 +101,22 @@ protected function setDefinition($id, Definition $definition)
         }
     }
 
-    private function findClasses($namespace, $pattern, $excludePattern)
+    private function findClasses($namespace, $pattern, array $excludePatterns)
     {
         $parameterBag = $this->container->getParameterBag();
 
         $excludePaths = array();
         $excludePrefix = null;
-        if ($excludePattern) {
-            $excludePattern = $parameterBag->unescapeValue($parameterBag->resolveValue($excludePattern));
-            foreach ($this->glob($excludePattern, true, $resource) as $path => $info) {
-                if (null === $excludePrefix) {
-                    $excludePrefix = $resource->getPrefix();
+        if ($excludePatterns) {
```

Comment:
you could remove this diff, handling the empty array like any other array (the `string|null` type was needing a dedicated handling of `null` before)